src/main/java/org/folio/notes/service/impl/NotesServiceImpl.java

```diff
@@ -149,13 +149,20 @@ public void updateLinks(String objectType, String objectId, NoteLinkUpdateCollec
       } else {
         noteEntity.addLink(linkEntity);
       }
+      if (noteEntity.getLinks().isEmpty()) {
+        noteRepository.delete(noteEntity);
+      } else {
+        noteRepository.save(noteEntity);
+      }
     }
-    noteRepository.saveAll(noteEntities);
   }
 
   @Transactional
   @Override
   public void updateNote(UUID id, Note dto) {
+    if (dto.getLinks().isEmpty()) {
+      deleteNote(id);
+    }
     noteRepository.findById(id)
```

Comment:
Shouldn't this step be skipped if the note has been deleted by the previous `if {}` statement?

Reply:
You are right, changed